doc/decisions/DR023-Versioning-traits-and-specifications.md

```diff
@@ -0,0 +1,702 @@
+  usage:
+    - entity
+  traitSet:
+    - namespace: usage
+      name: Entity
+    - namespace: application
+      name: Work
+    - namespace: content
+      name: LocatableContent
+```
+
+Conceptually, specifications are "things" that exhibit a set of traits.
+Current, specifications are purely unordered sets of `traitId`'s, raising similar problems as how to encode the version that individual traits have.
+
+Specifications are used to categorize entities when performing api operations. For example, they may be used to filter an asset browser to specific kinds of entity (Image, Video, TextDoc, etc). They are also used in publishing, where both host and manager have to agree on the "kind of thing" being published.
+
+Initially, one would want to version these things with the same
+mechanism to traits, with the version being updated when the `traitSet`
+of the specification is modified. However, what happens if a trait in
+the specification is merely versioned up, should this imply a
+specification schema version increment?
+
+Tentatively, no, or at least, it would be better if it didn't. The
+method of versioning plays into this decision quite strongly:
+
+### Versioning entire schema
+- Has the advantage that version bumps are always together, so if a
+  trait updates, the specification implicitly updates with it.
+- Much simpler to think about, the specification version in-play is
+  always the one the manager reports.
+- Don't have to find a way to pack specification version into `traitSet`
+- Makes discoverability extra difficult for managers that are trying to
+  support ranges of compatible versions with fewer branches.
+
+### Versioning specs individually
+- Raises the question of whether each trait in a specification needs to
+  be pinned to a specific version of a trait.
+- Have to pack specification version into the type, or ask the manager
+  to somehow deduce specification version
+
+The complexity of thinking about disconnected versioning when you take
+specifications into account does seem to bias towards versioning the
+entire schema as a unit.
+
+
+# What about multi-package
+The ecosystem and thinking around multi-package specifications isn't
+very developed yet, but nonetheless will be impacted by this decision.
+
+Consider another traits and specifications library alongside
+`openassetio-mediacreation` called `companycorp-traits`. You could build
+a specification :
+
+```yml
+ CompanyCorpCamera:
+    description: >
+      Any entity that holds three-dimensional data.
+    usage:
+      - entity
+    traitSet:
+      - namespace: openassetio_mediacreation.imaging.Camera
+        name: Camera
+      - namespace: companycorp-traits.identifier
+        name: CompanyCorp
+```
+
+This is troublesome to consider for a few reasons. It's tempting to just
+handwave this sort of thing away for later due to how complex it can
+get, but no, let's bite the bullet.
+
+## Explicit versioning
```

Comment:
[consider] Would like to see a worked example, similar to the "Whole-schema versioning" alternative below, since I suspect they'll result in similarly excessive branching.